### DISCUSSION TITLE
desktop and mobile packaging

### DISCUSSION THREAD

**Discussion by rraammiinn** [reflex-dev/reflex#1211] [General]
I don't think pynecone be able to translate any python code to js so it probably handles all events on backend and returns the results so it's costly for a web application but not for a desktop or mobile app so I think pynecone is more suitable for desktop and mobile apps where both server and client are on same machine and latency is quite low so why not use pynecone for these applications instead of web ?

**Comment by fasetto** [comment]
I was thinking the same thing. It would be great if we could create desktop apps with reflex-dev.

**Reply by Alek99** [reply]
Desktop apps will be easier to potentially implement in the future, with the use of something like electron.

**Comment by abcnow** [comment]
any plan on this in the near future?

**Comment by camstuart** [comment]
I think capacitor is the path to least resistance, see: https://github.com/orgs/reflex-dev/discussions/3152

**Comment by makerjackie** [comment]
wrap reflex to desktop app, two possible way
1. use pywebview: https://pywebview.flowrl.com/
2. use electron: https://github.com/gnoviawan/fast-api-electron-js

**Comment by lmbeauplet** [comment]
@Alek99 Hope your good ! Do we have news about desktop packaging following roadmap ? There are many solutions, but no example or proof of usage. I try to do this with tauri (rust) (instead of electron) and use the build command from tauri client. Solution can work, but I failed to achieve the goal and don't know if i take the better solution. 

I try Pyinstallr solution also but there are blocking issue that block the launching process after build. And so, with these things, we can't build a standalone executable with Pyinstaller, because, entrypoint that try to init and run search rx.config.py and search package named folder etc.

**Reply by itsmeadarsh2008** [reply]
I had done it.
https://discord.com/channels/1029853095527727165/1063735841333198938/1329062355966890025

**Reply by louismarieb** [reply]
Hi @itsmeadarsh2008 ! Thanks but I can't see the discussion on discord, can you share me you're solution by another way or here ?

**Reply by itsmeadarsh2008** [reply]
The message has a drive link which leads to a video where the demonstration of the idea is. This is a thread not a discussion.

**Reply by louismarieb** [reply]
On click, this link is redirected automatically to the base channel, and so I can't see video. Is there is another way to share me this ? I'm very interested and curious to see your solution (my discord is louma4817)

**Reply by itsmeadarsh2008** [reply]
Message Says:

https://drive.google.com/file/d/1g27IKgnPmuKt3FCWZIjFPJUyJETaSYvu/view?usp=sharing
Reflex is now a complete app framework. Integration of `pywebview` and measuring internet speeds locally. The file was too big so I had to upload it from a Google drive. Frameless Desktop App with Reflex. 

pywebview has support for Android app export using buildozer but we can't run fast API on Android however, we can run a static backend less site. I can't say about the iOS

**Reply by sandyowlet** [reply]
> @Alek99 Hope your good ! Do we have news about desktop packaging following roadmap ? There are many solutions, but no example or proof of usage. I try to do this with tauri (rust) (instead of electron) and use the build command from tauri client. Solution can work, but I failed to achieve the goal and don't know if i take the better solution.
> 
> I try Pyinstallr solution also but there are blocking issue that block the launching process after build. And so, with these things, we can't build a standalone executable with Pyinstaller, because, entrypoint that try to init and run search rx.config.py and search package named folder etc.

Hi, I just created a repository that demonstrates how to use pyinstaller and pywebview to package Reflex apps as executables:
https://github.com/sandyowlet/reflex-desktop-app
You can check the readme and learn how to use this example

**Comment by Alek99** [comment]
This is not currently on the roadmap right now we are focusing a lot on performance and modularity

**Comment by sandyowlet** [comment]
Hi everyone, I just created a repository that demonstrates how to use pyinstaller and pywebview to package Reflex apps as executables:
https://github.com/sandyowlet/reflex-desktop-app
You can check the readme and learn how to use this example
